### PR TITLE
Added goodie to convert alphabetic phone number to numeric phone number.

### PR DESCRIPTION
**Pull request by stevelippert**
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
Converts alphabetic phone numbers to numeric phone numbers. (E.g., Query: '1-800-fun-hack to phone number' returns 'Phone Number: 1-800-386-4225'

**What problem does your Instant Answer solve (Why is it better than organic links)?**
Right now organic results don't convert the alphabetic phone numbers into easy to dial numeric phone numbers.

**What is the data source for your Instant Answer? (Provide a link if possible)**
Mapped phone keys between numbers and the alphabet.

**Why did you choose this data source?**
Its the definitive source

**Are there any other alternative (better) data sources?**
None that I'm aware of.

**What are some example queries that trigger this Instant Answer?**
1-800-LAWYR-UP to digits
1-800-fun-hack to phone number

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Users in the United States that don't want to think about converting phone numbers with words / letters into an actual phone number.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes, https://duck.co/ideas/idea/4718/letters-to-keypad-numbers-and-vice-versa

**Which existing Instant Answers will this one supercede/overlap with?**

**Are you having any problems? Do you need our help with anything?**
I've only completed the first part, going from alphabetic phone numbers to numeric phone numbers, not going from numeric phone numbers to words.

**Where did you hear about DuckDuckHack? (For first time contributors)**
Blog entry, http://blog.ackx.net/duckduckhack-accepted.html.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
image
http://i.imgur.com/dbRbKDo.png

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
[X] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
[X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [X] Google Chrome
        [X] Firefox
        [] Opera
        [X] IE 10

    - Windows 7
        [X] Google Chrome
        [X] Firefox
        [] Opera
        [X] IE 8
        [X] IE 9
        [X] IE 10

    - Windows XP
        [] IE 7
        [X] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
```

### CODE REVIEW

File: lib/DDG/Goodie/PhoneAlphabet.pm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    #Return unless it looks like a phone number.
+    return unless ($_ =~ /[-0-9A-Za-z]{7,15}$/);
+    return unless $_;
+    #Lower case everything.
+    $_ = lc;
+    #Use a regex to replace each letter with the corresponding number from the phone key pad.
+    $_ =~ tr/abcdefghijklmnopqrstuvwxyz/22233344455566677778889999/;
+
+    #Returns the phone number.
+    return "Phone Number: $_";
```
Comment (jagtalon) [inline lib/DDG/Goodie/PhoneAlphabet.pm]: Would love to see the output as a structured answer https://duck.co/duckduckhack/goodie_advanced_handle_functions#using-structured-responses like https://duckduckgo.com/?q=1+%2B+1&ia=answer

File: lib/DDG/Goodie/PhoneAlphabet.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+attribution github => ["https://github.com/stevelippert", "Steve Lippert"],
+            twitter => ["https://twitter.com/stevelippert", "stevelippert"];
+
+# Triggers
+triggers any => 'to digit', 'to digits', 'to phone', 'to phone number';
+
+handle remainder => sub {
+    #Return unless it looks like a phone number.
+    return unless ($_ =~ /[-0-9A-Za-z]{7,15}$/);
+    return unless $_;
```
Comment (jagtalon) [inline lib/DDG/Goodie/PhoneAlphabet.pm]: @mintsoft Probably for something like +1-800

File: lib/DDG/Goodie/PhoneAlphabet.pm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+code_url "https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/PhoneAlphabet.pm";
+attribution github => ["https://github.com/stevelippert", "Steve Lippert"],
+            twitter => ["https://twitter.com/stevelippert", "stevelippert"];
+
+# Triggers
+triggers any => 'to digit', 'to digits', 'to phone', 'to phone number';
+
+handle remainder => sub {
+    #Return unless it looks like a phone number.
+    return unless ($_ =~ /[-0-9A-Za-z]{7,15}$/);
```
Comment (jagtalon) [inline lib/DDG/Goodie/PhoneAlphabet.pm]: @mintsoft Probably for something like +1-800

**Comment by jagtalon** [comment]
@stevelippert excited about this! @mintsoft I wonder if we'd need something like https://metacpan.org/pod/release/DCANTRELL/Number-Phone-3.0006/lib/Number/Phone.pm to verify phone numbers. It seems good enough to me though.

File: lib/DDG/Goodie/PhoneAlphabet.pm (hunk 1 above)
Comment (jagtalon) [inline lib/DDG/Goodie/PhoneAlphabet.pm]: @stevelippert what do you think about this?

**Comment by moollaza** [comment]
ping! @stevelippert are you still interested in working on this?

**Comment by stevelippert** [comment]
Sorry, I haven't had time to work on this lately and I think at this time I would have to say that I no longer have an interest in finishing this tool.

**Comment by moollaza** [comment]
@stevelippert I'm sorry to hear that you're no longer interested in working on it.  I'd love to understand why that is, so we can improve our PR handling process. Is it our fault? Did we take too long? We're you expecting this to be finished sooner? Feel free to email me if you prefer, moollaza@duckduckgo.com -- I'd appreciate any/all feedback you have!

FWIW, switching it to the Structured Answer mentioned above is a relatively simple, it's just a matter modifying the `return` so that a hash is given. I'll try and tackle that today so we can get this live for you shortly.

Thanks again for taking the time and effort to contribute, we really appreciate it!

**Comment by moollaza** [comment]
Closing. Continued in #964

**Comment by javathunderman** [comment]
Seeing as #964 was merged I think the "abandoned" label can be removed for this PR. @moollaza

**Pull request closed without merging**